src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -90,6 +103,28 @@ public override void OnStartActivity(Activity activity, object payload)
                 span.PutHttpUserAgentAttribute(userAgent);
                 span.PutHttpRawUrlAttribute(GetUri(request));
             }
+
+            if (newActivity != null)
+            {
+                if (newActivity.IsAllDataRequested)
+                {
+                    if (request.Host.Port == 80 || request.Host.Port == 443)
+                    {
+                        newActivity.AddTag("http.host", request.Host.Host);
+                    }
+                    else
+                    {
+                        newActivity.AddTag("http.host", request.Host.Host + ":" + request.Host.Port);
+                    }
+
+                    newActivity.AddTag("http.method", request.Method);
+                    newActivity.AddTag("http.path", path);
+
+                    var userAgent = request.Headers["User-Agent"].FirstOrDefault();
+                    newActivity.AddTag("http.user_agent", userAgent);
```

Comment:
It's possible for `FirstOrDefault` to return null so we should add null check too.

```suggestion
                    if (!string.IsNullOrEmpty(userAgent))
                    {                    
                        newActivity.AddTag("http.user_agent", userAgent);
                    }
```